src/components/BaseTextField/index.tsx

```diff
@@ -71,11 +71,34 @@ export interface BaseTextFieldProps extends BaseComponentProps {
   onMouseLeave?: ((event: React.MouseEvent<HTMLInputElement | HTMLTextAreaElement>) => void);
 }
 
+/**
+ * This is a hack to allow us to accept jsx elements in the description prop.
+ *
+ * We depend on onRenderDescription to display the description. However, the fabric
+ * textfield will only invoke it when it has a truthy description or errorMessage prop.
+ * Therefore we pass an arbitrary truthy string. It's value is only used to invoke
+ * onRenderDescription, and is otherwise ignored.
+ */
+const descriptionPresent = (props: BaseTextFieldProps) => {
+  if (props.description === undefined) {
+    return undefined;
+  }
+  return 'true';
+};
+
+const getOnRenderDescription = (description: string | JSX.Element | undefined, errorMessage: string | undefined) => {
+  if (!description || errorMessage) {
+    return () => null;
+  }
+  return () => <span className="y-base-text-field--description">{description}</span>;
+};
+
 export function getBaseTextFieldProps<T extends BaseTextFieldProps>(props: T): FabricTextFieldProps {
   return {
     value: props.value,
     label: props.label,
-    description: props.errorMessage ? undefined : props.description,
+    description: descriptionPresent(props),
+    onRenderDescription: getOnRenderDescription(props.description, props.errorMessage),
```

Comment:
getOnRenderDescription generates one of 2 new functions on each render cycle. instead of passing getOnRenderDescription, can we pass this.onRenderDescription which either returns null or the span with description?

Reply:
This is a bit of a challenge, since we need the `description` prop passed to YamUI's TextField, which we aren't able to pass into fabric's TextField because of the type mismatch.

Reply:
ah i see